ingenialink/servo.py

```diff
@@ -36,14 +36,16 @@ def __init__(self, servo):
 
     def run(self):
         """Checks if the drive is alive by reading the status word register"""
+        current_state = {}
+        for subnode in range(1, self.__servo.subnodes): 
+            current_state[subnode] = self.__servo.get_state(subnode)
         while not self.__stop:
             for subnode in range(1, self.__servo.subnodes):
                 try:
-                    status_word = self.__servo.read(
-                        self.__servo.STATUS_WORD_REGISTERS, subnode=subnode
-                    )
-                    state = self.__servo.status_word_decode(status_word)
-                    self.__servo._set_state(state, subnode=subnode)
+                    state = self.__servo.get_state(subnode)
+                    if current_state[subnode] != state:
```

Comment:
I think It would be clearer if the `current_state` dict is named `previous_states` and the `state`  variable is named `current_state`.